proposals/0268-main-menu-updating.md

```diff
@@ -12,171 +12,118 @@ This proposal seeks to add the ability to update the main menu with artworks and
-    </param>
+<function name="OnUpdateFile" functionID="OnUpdateFileID" messagetype="notification" since="x.x">
+    <description>This notification tells an app to upload and update a file with a given name.</description>
 
-    <param name="updateArtwork" type="Bool" mandatory="false">
-        <description>If not set, assume false. If true, the app should send a new AddCommand with the same cmdID as the previous one with the cmdIcon set. This will replace the UI cell with a new cell with the artwork displayed.</description>
+    <param name="fileName" type="String" maxlength="255" mandatory="true">
+        <description>File reference name.</description>
     </param>
 </function>
 ```
 
 ```xml
-<function name="UpdateSubMenu" functionID="UpdateSubMenuID" messagetype="notification" since="x.x">
+<function name="OnUpdateSubMenu" functionID="UpdateSubMenuID" messagetype="notification" since="x.x">
     <description>This notification tells an app to update the AddSubMenu or its 'sub' AddCommand and AddSubMenus with the requested data</description>
 
     <param name="menuID" type="Integer" minvalue="0" maxvalue="2000000000" mandatory="true">
         <description>This menuID must match a menuID in the current menu structure</description>
     </param>
 
-    <param name="updateArtwork" type="Bool" mandatory="false">
-        <description>If not set, assume false. If true, the app should send a new AddSubMenu with the same menuID as the previous one with the menuIcon set. This will replace the UI cell with a new cell with the artwork displayed. If subcells have already been sent, they should be transferred to the new AddSubmenu cell (and don't need to be re-sent).</description>
-    </param>
-
     <param name="updateSubCells" type="Bool" mandatory="false">
         <description>If not set, assume false. If true, the app should send AddCommands with parentIDs matching the menuID. These AddCommands will then be attached to the submenu and displayed if the submenu is selected.</description>
     </param>
 </function>
 ```
 
-#### Capabilities Updates
+#### Capabilities Updates (MOBILE_API + HMI_API)
 Updates will need to be made so that the app libraries and developers know if the head unit supports this feature or if they need to stick to the old-style of sending the entire menu at once.
 
 ```xml
-<struct name="WindowCapability" since="6.0">
-    <!-- New Parameters -->
-    <param name="supportsDynamicMenus" type="Bool" mandatory="false">
-        <description>If true, the head unit will send UpdateCommand and UpdateSubMenu notifications, and it will support updating menu cells by sending the same cmdID / menuID with new data. If not set, assume false.</description>
+<!-- New Struct -->
+<struct name="DynamicUpdateCapabilities" since="X.X">
+    <param name="supportedDynamicImageFieldNames" type="ImageFieldName" array="true" mandatory="false" minsize="1">
```

Comment:
Should this include `supportsDynamicMenus`, `supportsDynamicPerformInteractionImages`, and `supportsDynamicMenuImages`, as included in [this comment](https://github.com/smartdevicelink/sdl_evolution/issues/883#issuecomment-564263942)?

Reply:
No, because the steering committee decision was to give me the flexibility to create the capabilities that I thought would best work after assessing my comments, Kujtim's comments, and the various downfalls of both.